src/fs-client/index.ts

```diff
@@ -0,0 +1,105 @@
+import fs from "fs";
+import { createObjectCsvWriter } from "csv-writer";
+import { stringify } from "hjson";
+
+import {
+  BasePayment,
+  CycleReport,
+  DelegatorPayment,
+} from "src/engine/interfaces";
+import { get } from "lodash";
+
+const DELEGATOR_REPORT_HEADERS = [
+  { id: "cycle", title: "cycle" },
+  { id: "payment_type", title: "payment_type" },
+  { id: "delegator", title: "delegator" },
+  { id: "delegator_balance", title: "delegated_balance" },
+  { id: "total_baker_balance", title: "total_baker_balance" },
+  { id: "total_cycle_rewards", title: "total_cycle_rewards" },
+  { id: "fee_rate", title: "fee_rate" },
+  { id: "fee", title: "fee" },
+  { id: "amount", title: "amount" },
+  { id: "recipient", title: "recipient" },
+  { id: "tx_hash", title: "tx_hash" },
+  { id: "timestamp", title: "timestamp" },
+  { id: "note", title: "note" },
+];
```

Comment:
Do we need the baker balance, total reward? I feel like these should be in the hjson report for the cycle. 
Not sure about fee rate and fee. Maybe join it into one column? (like `0.001 (5%)`) I feel like there is too many columns 😄 

And please move timestamp at the beginning. It should be first imo.

Reply:
The vision here is to make things more easily auditable – e.g.  a delegator is able to calculate the payment amount based on cycle rewards, proportion of the delegated stake and fee rate.

Reply:
I see, but still the `baker balance`, `total reward` fields feel redundant. Users can check baker report before calculating 🤔 Eventually we can provide command for displaying just the audible part on demand.